src/main/java/io/kestra/plugin/microsoft365/sharepoint/Create.java

```diff
@@ -0,0 +1,180 @@
+package io.kestra.plugin.microsoft365.sharepoint;
+
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.slf4j.Logger;
+
+import com.microsoft.graph.models.DriveItem;
+import com.microsoft.graph.models.File;
+import com.microsoft.graph.core.ClientException;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Map;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Plugin(
+    examples = {
+        @io.kestra.core.models.annotations.Example(
+            title = "Create a new file in a SharePoint document library.",
+            code = {
+                "siteId: \"your-site-id\"",
+                "driveId: \"your-drive-id\"",
+                "parentId: \"your-parent-folder-id\"",
+                "filename: \"example.txt\"",
+                "content: \"File content here\""
+            }
+        )
+    }
+)
+@Schema(
+    title = "Create a new file or folder in a SharePoint document library.",
+    description = "This task allows you to create a new file or folder in a SharePoint document library."
+)
+public class Create extends Task implements RunnableTask<Create.Output> {
+    @Schema(
+        title = "The SharePoint site ID.",
+        description = "The unique identifier of the SharePoint site."
+    )
+    @PluginProperty(dynamic = true)
```

Comment:
No `@PluginProperty`, use just `Property<T>` as stated... in the contributor checklist ;-)